xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-webjar/src/main/webjar/lightbox.js

```diff
@@ -0,0 +1,293 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+define('lightboxTranslationKeys', {
+  prefix: 'core.viewers.attachments.',
+  keys: [
+    'date',
+    'author'
+  ]
+});
+
+require(['jquery', 'xwiki-meta', 'blueimp-gallery', 'xwiki-l10n!lightboxTranslationKeys', 'blueimp-gallery-fullscreen',
+    'blueimp-gallery-indicator'], function($, xm, gallery, l10n) {
+  var myOpenLightbox;
+  var cachedAttachments = {};
+
+  /*
+   * Make sure that the toolbar will remain open also while hovering it, not just the image.
+   */
+  var keepToolbarOpenOnHover = function(image, timeout) {
+    timeout = setTimeout(function() {
+      image.popover('hide');
+    }, 3000);
+    $('.popover').on('mouseenter', function() {
+      clearTimeout(timeout);
+    });
+    $('.popover').on('mouseleave', function() {
+      image.popover('hide');
+    });
+  };
+
+  /**
+   * Assign to each selected image a toolbar popover with download and lightbox options.
+   */
+  var enableToolbarPopovers = function() {
+    var timeout;
+    // Activate the lightbox for all images inside the xwikicontent. TODO: filter to consider only rendered images.
+    $('#xwikicontent img').popover({
```

Comment:
Have you tried the ``selector`` parameter that makes the popover work also with images that are added after this code is executed?

Reply:
Yes. But using the 'selector' parameter cannot actually be used with a 'manual' trigger, since I will need to select the images after anyway. But without the 'manual' trigger I cannot keep the popover open on hover (only if I add a timeout on it, which isn't nice IMO)